### PR TITLE
All claims fix homelessness mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
- Updates vets-json-schema version on vets-website to latest, which includes a small patch for the same homelessness POC name
- Allows veterans to type in whatever they want to the homelessness POC field
- Sanitizes veteran homelessness POC input in the submit transformer

## Testing done
- Tested locally
- Added test case

## Screenshots
N/A

## Acceptance criteria
- [x] don't validate veteran input on the homelessness POC name field
- [x] Ensure vets-api is only sent values that jive with the schema in vets-json-schema

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/pages/addDisabilities.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ export const uiSchema = {
               input => input.replace(/([^a-zA-Z0-9\-’.,/() ]+)/g, ''),
               // Get rid of extra whitespace characters
               input => input.trim(),
-              input => input.replace(/\s{2,}/, ' '),
+              input => input.replace(/\s{2,}/g, ' '),
```
Comment (markgreenburg) [inline src/applications/disability-benefits/all-claims/pages/addDisabilities.js]: Completely unrelated change, but I just noticed this tiny 'bug' on the 'Add Disabilities' autosuggest field and didn't feel like opening a new PR for it 😛

File: yarn.lock [hunk 2]
```diff
@@ -11963,9 +11963,9 @@ verror@1.3.6:
   dependencies:
     extsprintf "1.0.2"
 
-"vets-json-schema@https://github.com/department-of-veterans-affairs/vets-json-schema.git#0387db93d7fad4ac530afe0a5a4e5cd91cde8c99":
-  version "3.136.6"
-  resolved "https://github.com/department-of-veterans-affairs/vets-json-schema.git#0387db93d7fad4ac530afe0a5a4e5cd91cde8c99"
+"vets-json-schema@https://github.com/department-of-veterans-affairs/vets-json-schema.git#76083e33f175fb00392e31f1f5f90654d05f1fd2":
+  version "3.137.7"
```
Comment (markgreenburg) [inline yarn.lock]: [This PR](https://github.com/department-of-veterans-affairs/vets-json-schema/pull/336/files) changed the schema version from `3.136.6` to `3.137.6`, which is why the jump looks weird.

File: src/applications/disability-benefits/all-claims/pages/addDisabilities.js (hunk 1 above)
Comment (cvalarida) [reply src/applications/disability-benefits/all-claims/pages/addDisabilities.js]: Good catch!

File: src/applications/disability-benefits/all-claims/pages/homelessOrAtRisk.js [hunk 3]
```diff
@@ -144,6 +144,12 @@ export const schema = {
         otherAtRiskHousing,
       },
     },
-    homelessnessContact,
+    homelessnessContact: {
+      ...homelessnessContact,
+      properties: {
+        ...homelessnessContact.properties,
+        name: _.omit('pattern', homelessnessContact.properties.name),
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/pages/homelessOrAtRisk.js]: Our future selves might appreciate a comment on why we're intentionally ignoring the regex in the schema.

File: src/applications/disability-benefits/all-claims/tests/data/maximal-test.json [hunk 4]
```diff
@@ -17,7 +17,7 @@
       "needToLeaveHousing": true
     },
     "homelessnessContact": {
-      "name": "Name Here",
+      "name": "Name Here, and invalid chars with email elmer@fud.com",
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/tests/data/maximal-test.json]: :joy:

**Review by cvalarida — APPROVED** [review]

**Review by bshyong — APPROVED** [review]
LGTM

**Pull request merged**